frontend/app/src/utils/metamask.ts

```diff
@@ -26,10 +26,13 @@ export async function getMetamaskAddresses(): Promise<string[]> {
 
   assert(accountPermission);
 
-  const exposedAccounts = accountPermission.caveats.find(
-    caveat => caveat.name === 'exposedAccounts'
-  );
+  const addresses: string[] = [];
+  accountPermission.caveats.map(permission => {
```

Comment:
Since we are not using the result of the mapping here, we could replace map with a normal loop right?